test/benchmark/tests/conftest.py

```diff
@@ -0,0 +1,237 @@
+
+
+@pytest.fixture(scope="function")
+def vgg11_config_file_path(request):
+    return os.path.join(os.getcwd(), "tests", "suite", "vgg11.yaml")
+
+
+@pytest.fixture(scope="function")
+def vgg16_config_file_path(request):
+    return os.path.join(os.getcwd(), "tests", "suite", "vgg16.yaml")
+
+
+@pytest.fixture(scope="function")
+def bert_config_file_path(request):
+    return os.path.join(os.getcwd(), "tests", "suite", "bert.yaml")
+
+
+@pytest.fixture(scope="function")
+def mnist_config_file_path(request):
+    return os.path.join(os.getcwd(), "tests", "suite", "mnist.yaml")
+
+
+@pytest.fixture(scope="function")
+def fastrcnn_config_file_path(request):
+    return os.path.join(os.getcwd(), "tests", "suite", "fastrcnn.yaml")
+
+
+@pytest.fixture(scope="session")
+def region():
+    return os.getenv("AWS_DEFAULT_REGION", DEFAULT_REGION)
+
+
+@pytest.fixture(scope="session")
+def docker_client(region):
+    test_utils.run_subprocess_cmd(
+        f"$(aws ecr get-login --no-include-email --region {region})",
+        failure="Failed to log into ECR.",
+    )
+    return docker.from_env()
+
+
+@pytest.fixture(scope="session")
+def ecr_client(region):
+    return boto3.client("ecr", region_name=region)
+
+
+@pytest.fixture(scope="function")
+def ec2_key_name(request):
+    random.seed(datetime.datetime.now())
+    return f"{request.fixturename.replace('_', '-')}-{random.randrange(1, 10000, 3)}"
+
+
+@pytest.fixture(scope="session")
+def ec2_client(region):
+    return boto3.client("ec2", region_name=region, config=Config(retries={"max_attempts": 10}))
+
+
+@pytest.fixture(scope="session")
+def ec2_resource(region):
+    return boto3.resource("ec2", region_name=region, config=Config(retries={"max_attempts": 10}))
+
+
+@pytest.fixture(scope="function")
+def ec2_instance_type(request):
+    return request.param if hasattr(request, "param") else "c4.4xlarge"
+
+
+@pytest.fixture(scope="function")
+def ec2_instance_role_name(request):
+    return request.param if hasattr(request, "param") else IAM_INSTANCE_PROFILE
+
+
+@pytest.fixture(scope="function")
+def ec2_instance_ami(request):
+    return request.param if hasattr(request, "param") else AMI_ID
+
+
+@pytest.mark.timeout(300)
+@pytest.fixture(scope="function")
+def ec2_instance(
+    request,
+    ec2_client,
+    ec2_resource,
+    ec2_instance_type,
+    ec2_key_name,
+    ec2_instance_role_name,
+    ec2_instance_ami,
+    region,
+):
+    key_filename = ec2_utils.generate_ssh_keypair(ec2_client, ec2_key_name)
+
+    def delete_ssh_keypair():
+        ec2_utils.destroy_ssh_keypair(ec2_client, key_filename)
+
+    request.addfinalizer(delete_ssh_keypair)
+
+    params = {
+        "KeyName": ec2_key_name,
+        "ImageId": ec2_instance_ami,
+        "InstanceType": ec2_instance_type,
+        "IamInstanceProfile": {"Name": ec2_instance_role_name},
+        "TagSpecifications": [
+            {"ResourceType": "instance", "Tags": [{"Key": "Name", "Value": f"TS Benchmark {ec2_key_name}"}]},
```

Comment:
Can this be a config ?

Reply:
Do you mean it should be possible to chose a naming pattern for the instances? I think that makes sense, added to future improvements in this issue: https://github.com/pytorch/serve/issues/1067